code/game/objects/items/weapons/power_cells.dm

```diff
@@ -265,6 +270,9 @@
 		charge_rate = 0
 		damaged = FALSE //so you don't get the damaged examine if the cell is broken
 		desc = "The inner circuitry melted and the paint flaked off. It bulges slightly at the sides. <span class='warning'>It's going to explode any moment now.</span>"
+		for(var/mob/living/L in view(get_turf(src), 7)) //Oh, and irradiate everyone nearby in a bigger burst of radiation.
+			L.apply_radiation(initial(charge_rate)/5, RAD_EXTERNAL)
+		visible_message("<span class='warning'>\The [src] breaks, and has started leaking radioactive material!</span>")
```

Comment:
How would you know it's broken if it's inside something? Make it a description of sound or something "You hear sparks and sizzling coming from the [CELL/ITEM HOLDING THE CELL].

Reply:
I can try that, I'll need to check what proc it was that tells whether the item is contained in something or not.